subsys/bluetooth/controller/ll_sw/ull_conn.c

```diff
@@ -1506,12 +1506,18 @@ static inline void disable(u16_t handle)
 	LL_ASSERT(mark == conn);
 }
 
-static void conn_cleanup(struct ll_conn *conn)
+static void conn_cleanup(struct ll_conn *conn, u8_t reason)
 {
 	struct lll_conn *lll = &conn->lll;
 	struct node_rx_pdu *rx;
 	u32_t ticker_status;
 
+	/* Prepare the rx packet structure */
+	rx = (void *)&conn->llcp_terminate.node_rx;
+	rx->hdr.handle = conn->lll.handle;
+	rx->hdr.type = NODE_RX_TYPE_TERMINATE;
```

Comment:
It is not clear to me how the termination is propagated upwards from here. If the code in lines 1599ff is doing the propagation, maybe the respective code should be moved here to make it clearer?

Reply:
Only the termination structure is populated here, ULL context but enqueued in the LLL context in ```tx_lll_flush``` .The reason for population to void passing the ```reason``` and ```handle``` through the mayfly scheduling the tx_lll_flush.

Reply:
Ok, that makes sense. Maybe you could add what you just wrote as comment to the code?

Reply:
Will do.